catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -335,6 +365,75 @@ private void exportMetacard(File dumpLocation, Metacard metacard)
                 }
             }
         }
+
+        if (!ignoreContent) {
+            exportMetacardContent(metacard);
+        }
+    }
+
+    private void exportMetacardContent(Metacard metacard) {
+        Attribute attribute = metacard.getAttribute(Metacard.DERIVED_RESOURCE_URI);
+
+        if (attribute != null) {
+            List<Serializable> serializables = attribute.getValues();
+            serializables.forEach(serializable -> exportContentItem(serializable, metacard));
+        }
+
+        URI resourceUri = metacard.getResourceURI();
+        if (resourceUri != null) {
+            exportContentItem(resourceUri.toString(), metacard);
+        }
+
+    }
+
+    private void exportContentItem(Serializable serializable, Metacard metacard) {
+
+        Resource resource;
+        URI fileUri;
+
+        try {
+            fileUri = new URI((String) serializable);
+            ResourceRequest resourceRequest = new ResourceRequestByProductUri(fileUri);
+            ResourceResponse resourceResponse = getCatalog().getLocalResource(resourceRequest);
+            resource = resourceResponse.getResource();
+        } catch (URISyntaxException | InterruptedException | IOException | ResourceNotFoundException | ResourceNotSupportedException e) {
+            LOGGER.debug("Unable to retrieve content from metacard resource uri : {}",
+                    metacard.getResourceURI(),
+                    e);
+            return;
+        }
+
+        File contentDir = new File(contentPath);
+        if (!contentDir.mkdir()) {
+            LOGGER.warn("Unable to create content directory at {}.", dirPath);
+        }
+
+        String fileWritePath = contentPath;
+        if (StringUtils.isNotBlank(fileUri.getFragment())) {
+            fileWritePath = contentPath + fileUri.getFragment() + File.separator;
+            File fragmentDir = new File(fileWritePath);
+            if (!fragmentDir.mkdir()) {
+                LOGGER.warn("Unable to create fragment directory at {}.", fileWritePath);
+            }
+        }
+
+        try (InputStream inputStream = resource.getInputStream();
+                OutputStream outStream = new FileOutputStream(
+                        fileWritePath + metacard.getId() + "-" + resource.getName())) {
+
+            byte[] bytes = new byte[BUFFER_SIZE];
+
+            while (inputStream.read(bytes) != -1) {
```

Comment:
If read gets a partial buffer, then the write will output too much data. You should be keeping tracking of the number of bytes read and make sure you only write the same number.